src/test/java/com/commercetools/pingpong/model/GameTest.java

```diff
@@ -0,0 +1,312 @@
+package com.commercetools.pingpong.model;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.relaxng.datatype.DatatypeException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+
+public class GameTest {
+
+
+
+    public Player getDummyOne(int setScore, int matchScore, boolean serve, boolean beginningServe) {
+        Player dummyPlayerOne = new Player();
+
+        for (dummyPlayerOne.getSetScore(); dummyPlayerOne.getSetScore() < setScore;) {
+            dummyPlayerOne.addPoint();
+        }
+
+        for (dummyPlayerOne.getMatchScore(); dummyPlayerOne.getMatchScore() < matchScore;) {
+            dummyPlayerOne.addMatchScore();
+        }
+
+        if (serve) {
+            dummyPlayerOne.setServe();
+        } else {
+            dummyPlayerOne.unsetServe();
+        }
+
+        if (beginningServe) {
+            dummyPlayerOne.setBeginningServe();
+        } else {
+            dummyPlayerOne.unsetBeginningServe();
+        }
+
+        return dummyPlayerOne;
+    }
+
+
+    public Player getDummyTwo(int setScore, int matchScore, boolean serve, boolean beginningServe) {
+        Player dummyPlayerTwo = new Player();
+
+        for (dummyPlayerTwo.getSetScore(); dummyPlayerTwo.getSetScore() < setScore;) {
+            dummyPlayerTwo.addPoint();
+        }
+
+        for (dummyPlayerTwo.getMatchScore(); dummyPlayerTwo.getMatchScore() < matchScore;) {
+            dummyPlayerTwo.addMatchScore();
+        }
+
+        if (serve) {
+            dummyPlayerTwo.setServe();
+        } else {
+            dummyPlayerTwo.unsetServe();
+        }
+
+        if (beginningServe) {
+            dummyPlayerTwo.setBeginningServe();
+        } else {
+            dummyPlayerTwo.unsetBeginningServe();
+        }
+
+        return dummyPlayerTwo;
+    }
+
+
+
+
+    @Test
+    public void shouldReturnNewGame() {
+        Player testPlayerOne = new Player("One");
+        Player testPlayerTwo = new Player("Two");
+
+        Game testGame = new Game(testPlayerOne, testPlayerTwo);
+
+        assertThat("It should return the name of the first player", testGame.getPlayerOne().getName(), is("One"));
+        assertThat("It should return the name of the second player", testGame.getPlayerTwo().getName(), is("Two"));
+
+    }
+
+    @Test
+    public void shouldAddSetScoreOfPlayer() {
+        Message testMessage = new Message("1", 1);
+        Player testplayer = new Player();
+        Game testMatch = new Game(testplayer, testplayer);
+
+        testMatch.updateScoreOfPlayer(testplayer, testMessage.getActionType());
+
+        assertThat("", testplayer.getSetScore(), is(1));
+
+    }
+
+
+    /** WARNING: If any of the following five tests fail, the above dummyPlayers methods do not work properly **/
```

Comment:
That doesn't sound good 😅

Unit tests should be able to run in isolation and do not rely from each other. We need to fix the test setup in this case.

Reply:
We can have a look together.

Reply:
Gladly 😊